Feature/calendarreadonly

ScottBTR opened the pull request:
# Calendars Read-Only API

Calendars API for reading calendar/event information seamlessly across uwp, iOS and Android

# API

## Calendars

### Methods

| API | Description |
| ------------- | ------------- |
| ``Task<IEnumerable<Calendar>> GetCalendarsAsync()`` | Retrieve all calendars for device |
|  ``Task<IEnumerable<CalendarEvent>> GetEventsAsync( string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null)`` | Retrieve all events for a specified range/calendar. Null values for parameters will result in default values being used: null calendarId will return events from all calendares, startDate will default to the Now, endDate will default to startDate+14 days |
| ``Task<CalendarEvent> GetEventByIdAsync(string eventId)`` | Gets more detailed information about a specific event by the event id |

### Classes
```
public class Calendar
{
    string Id { get; set; }

    string Name { get; set; }
}

public class CalendarEvent
{
    string Id { get; set; }

    string CalendarId { get; set; }

    string Title { get; set; }

    string Description { get; set; }

    string Location { get; set; }

    bool AllDay { get; set; }

    DateTimeOffset Start { get; set; }

    DateTimeOffset? End { get; set; }

    TimeSpan? Duration { get; set; }

    IEnumerable<DeviceEventAttendee> Attendees { get; set; }
}

public class CalendarEventAttendee
{
    string Name { get; set; }

    string Email { get; set; }
}
```
# Scenarios

- User wants to see upcoming events.
- User wants to see who is attending an upcoming event.
- User wants to know when they will be free.

# Platform Compatibility
- Target Frameworks: <!-- All that apply -->
  - iOS:  Support on iOS for the API
  - Android: Support on Android for the API
  - UWP: Support on UWP for the API
# Backward Compatibility
- UWP: Windows 10 (introduced v10.0.10240.0) 
(https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.appointments.appointment)
- Android: android:minSdkVersion="14" 
(https://developer.android.com/reference/android/provider/CalendarContract)
- iOS: All versions until reminders are added, then iOS 6
(https://developer.apple.com/documentation/eventkit/ekeventstore)
# Difficulty : low

nickrandolph committed:
Added Calendar sample page with viewmodel
Added Calendar shared partial class
Added Calendar android partial class
Added Calendar (other platforms) partial class
Added IsSupported to all platforms
Added Calendar test for netstandard IsSupported

ScottBTR committed:
#1 - Allow Cheking and retrieving of permissions for android and iOS

Note:: For iOS there is no Calendar Read vs Write permissions, instead they have Calendar and Reminder permissions, which are quite different.
https://docs.microsoft.com/en-us/xamarin/ios/platform/eventkit

I've also noticed that although their is two seperate permissions for Calendar Read/Write for Android, when looking at the settings for the app after providing it with Read/Write Access it seems to set both as active, as there is no seperation in switches.

ScottBTR committed:
Allow Android to make a get calendars request.

NOTE:

Currently my CalendarObjects have a blocker where they cannot pull details from iOS or Android's provided libraries.

nickrandolph committed:
Fixing issue with disappearing properties on CalendarObject

nickrandolph committed:
Tidying up implementation code for Android

ScottBTR committed:
#2 Allow Xamarin to pull back a list of calendars for both iOS and Android

GetCalendarsAsync:
- Check/Request calendar read permissions
- Declare calendar instance/singleton
- Request List of iOS/Android platform specific calenders
- Cast to new shared List<DeviceCalendar> which obtains values for
  * Id - Platform specific android generally holds number ids whilst iOS uses guids
  * Name - Name of each calendar
  * IsReadOnly - Can this calendar be edited
- Currently the sample app only displays the name attribute, but we could indicate read only access and ids at a later state.

ScottBTR committed:
Amend changes set by PR for Issues - #1 #2 #3

ScottBTR committed:
Calendar Events able to be displayed for all or a specified calendar

ScottBTR committed:
Calendar Events able to be displayed for all or a specified calendar

ScottBTR committed:
Calendar Events able to be displayed for all or a specified calendar

ScottBTR committed:
Get Calendars working in sample for both all or specificed calendar on one page.

ScottBTR committed:
Get Calendars working in sample for both all or specificed calendar on one page.

ScottBTR committed:
Get Calendars working in sample for both all or specificed calendar on one page.

ScottBTR committed:
Reorder properties so that functionality comes after all properties.

ScottBTR committed:
Reorder properties so that functionality comes after all properties.

ScottBTR committed:
Reorder properties so that functionality comes after all properties.

ScottBTR committed:
Trial including start and end dates for event requests.

ScottBTR committed:
Trial including start and end dates for event requests.

ScottBTR committed:
Trial including start and end dates for event requests.

ScottBTR committed:
Allow Date & Time pickers to control event display (also works with specific calendars)

ScottBTR committed:
Allow Date & Time pickers to control event display (also works with specific calendars)

ScottBTR committed:
Allow Date & Time pickers to control event display (also works with specific calendars)

ScottBTR committed:
Revert Linker change.

ScottBTR committed:
Provide code reference point

ScottBTR committed:
Provide code reference point

ScottBTR committed:
Provide code reference point

ScottBTR committed:
Added Checkboxes to enable and disable start date/time and end date/time, Also a modal!

Note - Might need help working out why my EventToCommandBehavior isn't working for my checkboxes.

They just use their property {getters/setters}, which I could just use but it was suggested that I use the EventToCommand approach where possible.

You can also get a modal to appear giving you extra information about an event when tapping on an item within the ListView.

ScottBTR committed:
Added Checkboxes to enable and disable start date/time and end date/time, Also a modal!

Note - Might need help working out why my EventToCommandBehavior isn't working for my checkboxes.

They just use their property {getters/setters}, which I could just use but it was suggested that I use the EventToCommand approach where possible.

You can also get a modal to appear giving you extra information about an event when tapping on an item within the ListView.

ScottBTR committed:
Added Checkboxes to enable and disable start date/time and end date/time, Also a modal!

Note - Might need help working out why my EventToCommandBehavior isn't working for my checkboxes.

They just use their property {getters/setters}, which I could just use but it was suggested that I use the EventToCommand approach where possible.

You can also get a modal to appear giving you extra information about an event when tapping on an item within the ListView.

ScottBTR committed:
Added sort by datestart, changed listview to HasUnevenRows to make all event titles visible. altered date/time pickers to not use/need nullables. Including deleted events for the time being.

ScottBTR committed:
Added sort by datestart, changed listview to HasUnevenRows to make all event titles visible. altered date/time pickers to not use/need nullables. Including deleted events for the time being.

ScottBTR committed:
Added sort by datestart, changed listview to HasUnevenRows to make all event titles visible. altered date/time pickers to not use/need nullables. Including deleted events for the time being.

ScottBTR committed:
Added sort by datestart, changed listview to HasUnevenRows to make all event titles visible. altered date/time pickers to not use/need nullables. Including deleted events for the time being.

ScottBTR committed:
Added sort by datestart, changed listview to HasUnevenRows to make all event titles visible. altered date/time pickers to not use/need nullables. Including deleted events for the time being.

ScottBTR committed:
Added Github repository reference to class pulled from outside source.

ScottBTR committed:
Added Github repository reference to class pulled from outside source.

ScottBTR committed:
Added new CalendarEventAddPage as a modal to enter data required to create an event

ScottBTR committed:
Added new CalendarEventAddPage as a modal to enter data required to create an event

ScottBTR committed:
Added new CalendarEventAddPage as a modal to enter data required to create an event

ScottBTR committed:
Added new CalendarEventAddPage as a modal to enter data required to create an event

ScottBTR committed:
Added new CalendarEventAddPage as a modal to enter data required to create an event

ScottBTR committed:
Added new CalendarEventAddPage as a modal to enter data required to create an event

ScottBTR committed:
Add backend data call to be able to create new event when provided with a calendar id and required fields.

ScottBTR committed:
Add backend data call to be able to create new event when provided with a calendar id and required fields.

ScottBTR committed:
Add backend data call to be able to create new event when provided with a calendar id and required fields.

ScottBTR committed:
Add backend data call to be able to create new event when provided with a calendar id and required fields.

ScottBTR committed:
Add backend data call to be able to create new event when provided with a calendar id and required fields.

ScottBTR committed:
Add backend data call to be able to create new event when provided with a calendar id and required fields.

ScottBTR committed:
Clean up according to PR comments - Removal of not-needed converter EpocToDateTimeConverter

Removal of Reminders
Altered Recurrance variables to use the class RecurrancePattern (will require hooking up to database later on)

ScottBTR committed:
Clean up according to PR comments - Removal of not-needed converter EpocToDateTimeConverter

Removal of Reminders
Altered Recurrance variables to use the class RecurrancePattern (will require hooking up to database later on)

ScottBTR committed:
Clean up according to PR comments - Removal of not-needed converter EpocToDateTimeConverter

Removal of Reminders
Altered Recurrance variables to use the class RecurrancePattern (will require hooking up to database later on)

ScottBTR committed:
Clean up according to PR comments - Removal of not-needed converter EpocToDateTimeConverter

Removal of Reminders
Altered Recurrance variables to use the class RecurrancePattern (will require hooking up to database later on)

ScottBTR committed:
Clean up according to PR comments - Removal of not-needed converter EpocToDateTimeConverter

Removal of Reminders
Altered Recurrance variables to use the class RecurrancePattern (will require hooking up to database later on)

ScottBTR committed:
Clean up according to PR comments - Removal of not-needed converter EpocToDateTimeConverter

Removal of Reminders
Altered Recurrance variables to use the class RecurrancePattern (will require hooking up to database later on)

ScottBTR committed:
Remove Calendar Add Event feature [Android] - May be reimplemented at a later date.

ScottBTR committed:
Remove Calendar Add Event feature [Android] - May be reimplemented at a later date.

ScottBTR committed:
Remove Calendar Add Event feature [Android] - May be reimplemented at a later date.

ScottBTR committed:
Remove Calendar Add Event feature [Android] - May be reimplemented at a later date.

ScottBTR committed:
FiX calendareventpage to also use stringformat instead of EpochToDateTimeConverter (deprecated)

ScottBTR committed:
FiX calendareventpage to also use stringformat instead of EpochToDateTimeConverter (deprecated)

ScottBTR committed:
FiX calendareventpage to also use stringformat instead of EpochToDateTimeConverter (deprecated)

ScottBTR committed:
FiX calendareventpage to also use stringformat instead of EpochToDateTimeConverter (deprecated)

ScottBTR committed:
Bring back backing fields, or else fields outside the checkbox are not recognizing the change in value, i.e. the date/time pickers isenabled is not being invoked.

ScottBTR committed:
Bring back backing fields, or else fields outside the checkbox are not recognizing the change in value, i.e. the date/time pickers isenabled is not being invoked.

ScottBTR committed:
Bring back backing fields, or else fields outside the checkbox are not recognizing the change in value, i.e. the date/time pickers isenabled is not being invoked.

ScottBTR committed:
Bring back backing fields, or else fields outside the checkbox are not recognizing the change in value, i.e. the date/time pickers isenabled is not being invoked.

ScottBTR committed:
Made changes, addressing PR feedback

ScottBTR committed:
Made changes, addressing PR feedback

ScottBTR committed:
Made changes, addressing PR feedback

ScottBTR committed:
Made changes, addressing PR feedback

ScottBTR committed:
Moved comment to top of file (above usings)

ScottBTR committed:
Moved comment to top of file (above usings)

ScottBTR committed:
Moved comment to top of file (above usings)

ScottBTR committed:
Grids within Grids don't seem to work properly, have tried using verticaloptions to manipulate rather than set values but to no avail, will keep the height request within grid instead of buttons for now.

ScottBTR committed:
Grids within Grids don't seem to work properly, have tried using verticaloptions to manipulate rather than set values but to no avail, will keep the height request within grid instead of buttons for now.

ScottBTR committed:
Grids within Grids don't seem to work properly, have tried using verticaloptions to manipulate rather than set values but to no avail, will keep the height request within grid instead of buttons for now.

ScottBTR committed:
Grids within Grids don't seem to work properly, have tried using verticaloptions to manipulate rather than set values but to no avail, will keep the height request within grid instead of buttons for now.

ScottBTR committed:
Addressing PR comment regarding reuse of static string.

ScottBTR committed:
Addressing PR comment regarding reuse of static string.

ScottBTR committed:
Addressing PR comment regarding reuse of static string.

ScottBTR committed:
Addressing PR comment regarding reuse of static string.

ScottBTR committed:
Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work..

ScottBTR committed:
Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work..

ScottBTR committed:
Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work..

ScottBTR committed:
Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work..

ScottBTR committed:
Revert "Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work.."

This reverts commit 973df83ef71b0d461d4e12b1a8146f8b42f989e9.

ScottBTR committed:
Revert "Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work.."

This reverts commit 973df83ef71b0d461d4e12b1a8146f8b42f989e9.

ScottBTR committed:
Revert "Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work.."

This reverts commit 973df83ef71b0d461d4e12b1a8146f8b42f989e9.

ScottBTR committed:
Revert "Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work.."

This reverts commit 973df83ef71b0d461d4e12b1a8146f8b42f989e9.

ScottBTR committed:
Revert "Revert "Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work..""

This reverts commit 751f05d127c7a1e81d675c2977db3aeeb61e3e19.

ScottBTR committed:
Revert "Revert "Working out pulling back events for ios for any and specific calendars, currently NSDate to Epoch conversion still needs work..""

This reverts commit 751f05d127c7a1e81d675c2977db3aeeb61e3e19.

ScottBTR committed:
Adjustments made to meet PR requests

ScottBTR committed:
Adjustments made to meet PR requests

ScottBTR committed:
Adjustments made to meet PR requests

ScottBTR committed:
Adjustments made to meet PR requests

ScottBTR committed:
iOS requesting events now works with date from/to, also sorts the resulting list by start date.

ScottBTR committed:
iOS requesting events now works with date from/to, also sorts the resulting list by start date.

ScottBTR committed:
iOS requesting events now works with date from/to, also sorts the resulting list by start date.

ScottBTR committed:
iOS requesting events now works with date from/to, also sorts the resulting list by start date.

ScottBTR committed:
fix bug that occurs when end date is updated it is resulting in a null end date instead of returning the assigned end date.

ScottBTR committed:
fix bug that occurs when end date is updated it is resulting in a null end date instead of returning the assigned end date.

ScottBTR committed:
fix bug that occurs when end date is updated it is resulting in a null end date instead of returning the assigned end date.

ScottBTR committed:
fix bug that occurs when end date is updated it is resulting in a null end date instead of returning the assigned end date.

ScottBTR committed:
Alter Description to be iOS 'Notes' as 'Description' is actually the contents of the class.

Description: https://developer.apple.com/documentation/objectivec/nsobject/1418799-description

Notes:
https://developer.apple.com/documentation/eventkit/ekcalendaritem/1507507-notes

At a later date I should also probably use 'hasNotes' to return a more suitable message, or even as a lower level version of the event, that allows the user to determine whether they should request further data. (same with reminders, alarms and so forth)

ScottBTR committed:
Alter Description to be iOS 'Notes' as 'Description' is actually the contents of the class.

Description: https://developer.apple.com/documentation/objectivec/nsobject/1418799-description

Notes:
https://developer.apple.com/documentation/eventkit/ekcalendaritem/1507507-notes

At a later date I should also probably use 'hasNotes' to return a more suitable message, or even as a lower level version of the event, that allows the user to determine whether they should request further data. (same with reminders, alarms and so forth)

ScottBTR committed:
Alter Description to be iOS 'Notes' as 'Description' is actually the contents of the class.

Description: https://developer.apple.com/documentation/objectivec/nsobject/1418799-description

Notes:
https://developer.apple.com/documentation/eventkit/ekcalendaritem/1507507-notes

At a later date I should also probably use 'hasNotes' to return a more suitable message, or even as a lower level version of the event, that allows the user to determine whether they should request further data. (same with reminders, alarms and so forth)

ScottBTR committed:
Alter Description to be iOS 'Notes' as 'Description' is actually the contents of the class.

Description: https://developer.apple.com/documentation/objectivec/nsobject/1418799-description

Notes:
https://developer.apple.com/documentation/eventkit/ekcalendaritem/1507507-notes

At a later date I should also probably use 'hasNotes' to return a more suitable message, or even as a lower level version of the event, that allows the user to determine whether they should request further data. (same with reminders, alarms and so forth)

ScottBTR committed:
Using PushAsync instead of PushModalAsync as an easy way to allow back button to be present (for iOS)

ScottBTR committed:
Using PushAsync instead of PushModalAsync as an easy way to allow back button to be present (for iOS)

ScottBTR committed:
Using PushAsync instead of PushModalAsync as an easy way to allow back button to be present (for iOS)

ScottBTR committed:
Using PushAsync instead of PushModalAsync as an easy way to allow back button to be present (for iOS)

ScottBTR committed:
Small text fix ups.

ScottBTR committed:
Small text fix ups.

ScottBTR committed:
Small text fix ups.

ScottBTR committed:
Small text fix ups.

ScottBTR committed:
don't need to await read access from viewmodel, each request already calls appropriate access

ScottBTR committed:
don't need to await read access from viewmodel, each request already calls appropriate access

ScottBTR committed:
don't need to await read access from viewmodel, each request already calls appropriate access

ScottBTR committed:
don't need to await read access from viewmodel, each request already calls appropriate access

ScottBTR committed:
Added GetEventByIdAsycn --> Works at [Android] and [iOS] level currently.

ScottBTR committed:
Added GetEventByIdAsycn --> Works at [Android] and [iOS] level currently.

ScottBTR committed:
Default EndTime to 11:59pm so that if you turn both start date and end date on it gives you todays events, rather than starting on the same time resulting in always 0 results.

Perhaps I should store the string as a const somewhere :thinking:

ScottBTR committed:
Default EndTime to 11:59pm so that if you turn both start date and end date on it gives you todays events, rather than starting on the same time resulting in always 0 results.

Perhaps I should store the string as a const somewhere :thinking:

ScottBTR committed:
[Android] Attendees can now be retrieved and rules are about half way. need to work out how the monthly and weekly list work from flutter.

ScottBTR committed:
[Android] Attendees can now be retrieved and rules are about half way. need to work out how the monthly and weekly list work from flutter.

ScottBTR committed:
Don't break when BYDAY= is the last parameter, Should add this on all parameters.

Must also first create the list before adding days to it.

ScottBTR committed:
Don't break when BYDAY= is the last parameter, Should add this on all parameters.

Must also first create the list before adding days to it.

ScottBTR committed:
Checks to make sure rules are non-nullable to prevent an unnessacary/crash from substringing.

ScottBTR committed:
Checks to make sure rules are non-nullable to prevent an unnessacary/crash from substringing.

ScottBTR committed:
UWP working with requesting read/write, pulling back calendares, pulling back events for calendars or a specific calendars, with time constraints. as well as event details.

NOTE: may need to work out a singleton pattern.

ScottBTR committed:
UWP working with requesting read/write, pulling back calendares, pulling back events for calendars or a specific calendars, with time constraints. as well as event details.

NOTE: may need to work out a singleton pattern.

ScottBTR committed:
iOS Attendee Calendars

ScottBTR committed:
iOS Attendee Calendars

ScottBTR committed:
Don't try to build attendee list when there are none.

ScottBTR committed:
Don't try to build attendee list when there are none.

ScottBTR committed:
Allow Android to hold Deleted bool for Calendar and Event, so that users can decide whether to show those instances or not. Perhaps I need to make an android specific calendar extension hmmm

ScottBTR committed:
Allow Android to hold Deleted bool for Calendar and Event, so that users can decide whether to show those instances or not. Perhaps I need to make an android specific calendar extension hmmm

ScottBTR committed:
Add tests, and use default end date of 14 days away (there is a function that can override this setting)

ScottBTR committed:
Add tests, and use default end date of 14 days away (there is a function that can override this setting)

ScottBTR committed:
iOS Hot.Reload and Hot.Restart

mrlacey committed:
Fix broken docs link (#971)

Can't have a relative path to the wiki here

mattleibow committed:
Merge remote-tracking branch 'refs/remotes/origin/dev/1.4.0'

ScottBTR committed:
Addressing PR changes

ScottBTR committed:
Addressing PR changes

ScottBTR committed:
Creating Extension Methods

ScottBTR committed:
Creating Extension Methods

ScottBTR committed:
Merge branch 'feature/calendar' into feature/requesteventuniquely

ScottBTR committed:
Allow Uwp To see appointments, fix android trying to throw SQLException from string id.

ScottBTR committed:
Allow Uwp To see appointments, fix android trying to throw SQLException from string id.

ScottBTR committed:
Merge branch 'feature/calendartests' into feature/inclusionofandroidaddevent

ScottBTR committed:
Code cleanup to allow merge to be functional

ScottBTR committed:
Code cleanup to allow merge to be functional

ScottBTR committed:
Create event with provided values for android and return id of event created.

ScottBTR committed:
Create event with provided values for android and return id of event created.

ScottBTR committed:
Have Calendar creation (somewhat) working with creation, has strange permission issues that will need to be resolved.

ScottBTR committed:
Have Calendar creation (somewhat) working with creation, has strange permission issues that will need to be resolved.

mattleibow committed:
Update FileProvider_Tests.cs

mattleibow committed:
Update AppDelegate.cs

ScottBTR committed:
Create Event for iOS

ScottBTR committed:
Create Event for iOS

ScottBTR committed:
Create Event requires string because iOS uses guids.

ScottBTR committed:
Create Event requires string because iOS uses guids.

ScottBTR committed:
Addressing PR Feedback

ScottBTR committed:
Addressing PR Feedback

ScottBTR committed:
Revert not needed csproj changes to address PR feedback.

ScottBTR committed:
Revert not needed csproj changes to address PR feedback.

ScottBTR committed:
Merge pull request #30 from builttoroam/feature/requesteventuniquely

Feature/requesteventuniquely

ScottBTR committed:
Merge branch 'feature/calendar' into feature/uwpintegration

ScottBTR committed:
Merge branch 'feature/requesteventuniquely' into feature/uwpintegration

ScottBTR committed:
Spacing issue, damn you style cop!

ScottBTR committed:
Spacing issue, damn you style cop!

ScottBTR committed:
Address PR Comments

ScottBTR committed:
Address PR Comments

ScottBTR committed:
Addressing PR Comment in relation to not disposing of Cursor.

ScottBTR committed:
Addressing PR Comment in relation to not disposing of Cursor.

mattleibow committed:
Getting the solution to load without freezing + updates (#977)

jamesmontemagno committed:
Test, Sample, & Code Cleanup (#960)

* Fix Geocoding & Sensor Tests

* Bump version numbers for device tetss

* Add unit converters to samples & bump samples sdk

* do grpc checks for android

* Use new C# features

* Typo... Can't have the metric looking bad ;)

pictos committed:
GH-968 Fix AtmospheresToPascals and PascalsToAtmospheres methods (#969)

* Fixed Atmosphere to Pascals converter

* Added unit tests

* Changed LangVer to 8.0

* Changed to C#7.3

* Update UnitConverters_Tests.cs

* Fix spacing

munkii committed:
Add Weight conversion functionality (#983)

* Add Weight conversion functionality similar to the distance and other Unit Conversion methods
Add Unit Tests for these new conversions

* Delete Resource.designer.cs

* Add reference docs for unit converters.

mrlacey committed:
Add `ColorConverters.GetCompliment()` method (#970)

* Add ColoConverter.GetCompliment method

+ tests & docs
For #957

* Add ColoConverter.GetCompliment method

+ tests & docs
For #957

* Update index files

ScottBTR committed:
Adjusted to PR Comments, removed double up of Description

ScottBTR committed:
Adjusted to PR Comments, removed double up of Description

ScottBTR committed:
Change Cursor to using for auto dispose purposes.

ScottBTR committed:
Change Cursor to using for auto dispose purposes.

ScottBTR committed:
Allow AllDay to turn on/off time pickers, as if it's an all day event the time slots aren't required.

ScottBTR committed:
Allow AllDay to turn on/off time pickers, as if it's an all day event the time slots aren't required.

ScottBTR committed:
Add missing AllDay parameters for ios and uwp

ScottBTR committed:
Add missing AllDay parameters for ios and uwp

ScottBTR committed:
Disable AddEvent button unless an event has a proper start/end date and an event name

ScottBTR committed:
Disable AddEvent button unless an event has a proper start/end date and an event name

ScottBTR committed:
Adjusted according to PR comments

ScottBTR committed:
Adjusted according to PR comments

ScottBTR committed:
Moved Content into a scroll view incase either there are too many attendees or the description is too long.

ScottBTR committed:
Moved Content into a scroll view incase either there are too many attendees or the description is too long.

ScottBTR committed:
Merge pull request #31 from builttoroam/feature/uwpintegration

Feature/uwpintegration + Attendee list retrieval (+rules for android but that be pushed back until a later date)

ScottBTR committed:
Merge branch 'feature/calendar' into feature/calendartests

ScottBTR committed:
Merge branch 'feature/calendartests' into feature/inclusionofandroidaddevent

ScottBTR committed:
Fixes to bad auto-merge plus add the ability to change the defaultstartingtimefromnow

ScottBTR committed:
Fixes to bad auto-merge plus add the ability to change the defaultstartingtimefromnow

ScottBTR committed:
Prepare Calendar API for readonly release

ScottBTR committed:
Prepare Calendar API for readonly release

ScottBTR committed:
Remove android rule getting, as it's not required for this release, and make sure to move to the first instance of the cursor or nothing will/can be retrieved.

ScottBTR committed:
Remove android rule getting, as it's not required for this release, and make sure to move to the first instance of the cursor or nothing will/can be retrieved.

ScottBTR committed:
Remove not-needed files, fix formatting of xaml files

ScottBTR committed:
Remove not-needed files, fix formatting of xaml files

ScottBTR committed:
NullReferenece Exception to match ios/android catch/behaviour

ScottBTR committed:
NullReferenece Exception to match ios/android catch/behaviour

ScottBTR committed:
Revert changes made for hot.restart functionality

ScottBTR committed:
Move one user string/int to consts for better readability

ScottBTR committed:
Move one user string/int to consts for better readability

ScottBTR committed:
Remove Interfaces, Remove Deleted property from calendar & event (query will not include them at this stage for android), Change Start & End to StartDate & EndDate type long --> DateTimeOffset

ScottBTR committed:
Remove Interfaces, Remove Deleted property from calendar & event (query will not include them at this stage for android), Change Start & End to StartDate & EndDate type long --> DateTimeOffset

ScottBTR committed:
IReadOnlyList --> List as we've decided not to move interfaces going foward so class/return types should match.

ScottBTR committed:
IReadOnlyList --> List as we've decided not to move interfaces going foward so class/return types should match.

ScottBTR committed:
Alter Calendar Event/Attendee names to be more specific (and not existing c# words, see 'event')

ScottBTR committed:
Alter Calendar Event/Attendee names to be more specific (and not existing c# words, see 'event')

TianqiZhang committed:
Move to SDP pipeline (#998)

jamesmontemagno committed:
Add Android Platform code for activity/context (#997)

* Add Android Platform code for activity/context
* Sample code & docs
* Expose GetCurrentUIViewController

Redth committed:
Fix path to reference assemblies on win for mdoc (#1001)

ScottBTR committed:
Merge branch 'feature/calendarreadonly' of https://github.com/builttoroam/Essentials into feature/calendarreadonly

ScottBTR committed:
Revert changes used for debugging purposes.

ScottBTR committed:
Allow iOS tests to work (updating info.plist), make android exception consistent, and add multiple test cases

nickrandolph commented on Samples/Samples.Android/Properties/AndroidManifest.xml:
```diff
@@ -8,7 +8,9 @@
 	<uses-permission android:name="android.permission.CAMERA" />
 	<uses-permission android:name="android.permission.FLASHLIGHT" />
 	<uses-permission android:name="android.permission.INTERNET" />
+  <uses-permission android:name="android.permission.READ_CALENDAR" />
```
Can we move the added permissions so that they're next to each other

ScottBTR replied:
Ahh I see, it lines up when you edit it in xml but the difference is spaces vs tabs, fixing now.

nickrandolph replied:
No I meant move read and write permissions to consecutive lines

ScottBTR replied:
Sure, I only had them that way as they seemed to be listing permissions in alphabetical order

nickrandolph replied:
ah good point, I hadn't picked up on that. Leave as is then

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.android.cs:
```diff
@@ -0,0 +1,179 @@
+            };
+            var queryConditions = $"{CalendarContract.Calendars.InterfaceConsts.Deleted} != 1";
+
+            var cur = Platform.AppContext.ApplicationContext.ContentResolver.Query(calendarsUri, calendarsProjection.ToArray(), queryConditions, null, null);
```
does  "cur" implement IDisposable, if so, we should use a using statement here

ScottBTR replied:
Altered to match GetEventById

nickrandolph replied:
This one still needs a using statement

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.shared.cs:
```diff
@@ -0,0 +1,29 @@
+
+        static TimeSpan defaultEndTimeFromStartTime = TimeSpan.FromDays(14);
+
+        public static bool IsSupported => PlatformIsSupported;
```
I think we need to remove this based on the feedback on github

ScottBTR replied:
Removed

ScottBTR committed:
Remove Platform --> IS Supported, and move android cursor to using.

ScottBTR committed:
SetupForDuration field to be added

ScottBTR committed:
Adjust times to work with your timezone cross platform.

Note: There is one slight difference between android vs ios/uwp

Android uses the start time as a 'hard' start, i.e. if you check your schedule at 9:33 when you appointment was 9:30-9:40 it won't show up, whilst for ios/UWP it will show that appointment until 9:40.

ScottBTR committed:
Update the android query to replicate the behaviour of UWP/iOS

nickrandolph commented on Xamarin.Essentials/Types/DeviceCalendar.shared.cs:
```diff
@@ -30,6 +30,8 @@ public class DeviceEvent
 
         public DateTimeOffset? StartDate { get; set; }
 
+        public TimeSpan Duration { get; set; }
```
I don't think we should have three, individually set properties. Either EndDate or Duration should be a calculated value

ScottBTR replied:
Dealt with in later PR

ScottBTR committed:
Calendar no longer retrieves duration, but now calculates it based on start/end date, when either a start or end date is missing duration will return 0 as it's unknown, could be nullable otherwise?

ScottBTR committed:
Add Setter to Duration, I think we're okay to not make start date and end date nullable as allday just makes the times irrelevant but still needs to know the day it occurs. will test cross platform.

ScottBTR committed:
Convert List --> IEnumerables, Issue #35

ScottBTR committed:
Revert Explicit change as it's unnecessary

ScottBTR committed:
Convert AllDay to a getter and setter so that it better aligns with enddate/duration. Since for all day events we only need to know what date it's on.

ScottBTR committed:
Merge pull request #39 from builttoroam/bug/calendarListToIEnum

Bug/calendar list to IEnum

ScottBTR committed:
iOS and UWP to calculate EndDate based on AllDay for consistency with Android

ScottBTR committed:
Meet PR Feedback

ScottBTR committed:
Merge pull request #34 from builttoroam/feature/calendardurationgetter

Calendar no longer retrieves duration, but now calculates it based on…

ScottBTR committed:
Merge branch 'feature/calendarreadonly' into bug/calendarRemoveIsReadOnly

ScottBTR committed:
Removal of IsReadOnly from calendar

ScottBTR committed:
Remove IsReadOnly from test, Make duration align to center for better readability

ScottBTR committed:
Remove Default setters

ScottBTR committed:
Remove Calendar Read/Write Request access as they will be dealt with in next permission release

ScottBTR committed:
Merge pull request #40 from builttoroam/bug/calendarRemoveIsReadOnly

Bug/calendar remove is read only

ScottBTR committed:
Merge pull request #41 from builttoroam/bug/calendarremovedefaultdates

Remove Default setters

nickrandolph commented on DeviceTests/DeviceTests.iOS/Info.plist:
```diff
@@ -55,5 +55,9 @@
 	<string>Access to your location is required for cool things to happen!</string>
+  <key>NSCalendarsUsageDescription</key>
+  <string>Access to view and create calendar events</string>
+  <key>NSRemindersUsageDescription</key>
```
Can we remove NSRemindersUsageDescription and confirm whether we can still read the calendar events

ScottBTR replied:
can be removed addressed in: https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Samples/Samples.iOS/Info.plist:
```diff
@@ -55,6 +55,10 @@
 	<string>Access to your location is required for cool things to happen!</string>
+    <key>NSCalendarsUsageDescription</key>
+    <string>Access to view and create calendar events</string>
+    <key>NSRemindersUsageDescription</key>
```
remove NSRemindersUsageDescription and test

ScottBTR replied:
Addressed in: https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Samples/Samples/View/CalendarEventPage.xaml.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace Samples.View
+{
+    public partial class CalendarEventPage : BasePage
```
Check the code behind for other pages and see whether they specify the base class (ie BasePage). My preference is to remove this from the codebehind but we need to be consistent with what's on other pages

ScottBTR replied:
All other pages seem to inherit from BasePage

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.android.cs:
```diff
@@ -0,0 +1,159 @@
+                CalendarContract.Attendees.InterfaceConsts.AttendeeName
+            };
+            var attendeeSpecificAttendees = $"{CalendarContract.Attendees.InterfaceConsts.EventId}={eventId}";
+            var cur = Platform.AppContext.ApplicationContext.ContentResolver.Query(attendeesUri, attendeesProjection.ToArray(), attendeeSpecificAttendees, null, null);
```
Add using statement

ScottBTR replied:
Addressed in: https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.ios.cs:
```diff
@@ -0,0 +1,127 @@
+            {
+                throw new Exception($"iOS: Unexpected null reference exception {ex.Message}");
+            }
+            var calendarList = new List<DeviceCalendar>();
```
Change to:
var calendarList = (from calendar in calendars
                              select new DeviceCalendar
                {
                    Id = t.CalendarIdentifier,
                    Name = t.Title
                }).ToList();

ScottBTR replied:
Addressed in: https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.ios.cs:
```diff
@@ -0,0 +1,127 @@
+            var query = CalendarRequest.Instance.PredicateForEvents(sDate, eDate, calendars);
+            var events = CalendarRequest.Instance.EventsMatching(query);
+
+            foreach (var e in events)
```
See previous comment on using linq

nickrandolph replied:
Also, apply the sort condition as part of the linq expression

ScottBTR replied:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.ios.cs:
```diff
@@ -0,0 +1,127 @@
+        {
+            var attendees = new List<DeviceEventAttendee>();
+
+            foreach (var attendee in inviteList)
```
linq

ScottBTR replied:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

Added sort by name aswell

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.uwp.cs:
```diff
@@ -0,0 +1,122 @@
+            var instance = await CalendarRequest.GetInstanceAsync();
+            var uwpCalendarList = await instance.FindAppointmentCalendarsAsync(FindAppointmentCalendarsOptions.IncludeHidden);
+            var calendars = new List<DeviceCalendar>();
+            foreach (var c in uwpCalendarList)
```
linq

ScottBTR replied:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.uwp.cs:
```diff
@@ -0,0 +1,122 @@
+            var instance = await CalendarRequest.GetInstanceAsync();
+            var events = await instance.FindAppointmentsAsync(sDate, eDate.Subtract(sDate), options);
+            var eventList = new List<DeviceEvent>();
+            foreach (var e in events)
```
linq - use where clause and orderby for sorting

ScottBTR replied:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Calendar/Calendar.uwp.cs:
```diff
@@ -0,0 +1,122 @@
+        {
+            var attendees = new List<DeviceEventAttendee>();
+
+            foreach (var attendee in inviteList)
```
linq :-)

ScottBTR replied:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Permissions/Permissions.android.cs:
```diff
@@ -80,6 +80,7 @@ static Task<PermissionStatus> PlatformCheckStatusAsync(PermissionType permission
         static async Task<PermissionStatus> PlatformRequestAsync(PermissionType permission)
         {
             // Check status before requesting first
+
```
Roll back changes to this file

nickrandolph commented on Xamarin.Essentials/Permissions/Permissions.shared.enums.cs:
```diff
@@ -23,11 +23,14 @@ enum PermissionType
         LocationWhenInUse,
         Maps,
         NetworkState,
+        Reminders,
```
Do we need this permission?

ScottBTR replied:
Nope Addressed in:
https://github.com/builttoroam/Essentials/pull/43

nickrandolph commented on Xamarin.Essentials/Permissions/Permissions.shared.enums.cs:
```diff
@@ -23,11 +23,14 @@ enum PermissionType
         Battery,
         Camera,
+        CalendarRead,
+        CalendarWrite,
```
Remove write permissions changes

ScottBTR replied:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43

ScottBTR committed:
Project no longer requires buttons/test cases

ScottBTR committed:
PR Permissions update

ScottBTR committed:
PR Linq changes + using(cur) + aligning Location

ScottBTR committed:
Remove not needed WRITE_CALENDAR permissions from android at this stage.

ScottBTR committed:
Add back in <string>UIInterfaceOrientationPortraitUpsideDown</string> to DeviceTest.ios info.plist

ScottBTR committed:
Merge pull request #43 from builttoroam/bug/calendarLinq&PermissionChanges

Bug/calendar linq&permission changes

ScottBTR committed:
Merge pull request #42 from builttoroam/bug/calendarAlterSampleToRemoveRequestButtons

Project no longer requires buttons/test cases

nickrandolph committed:
Removing read and write commands
